### PR TITLE
Convert codebase to ESM

### PR DESCRIPTION
**Pull request by digitalsadhu**

### CODE REVIEW

File: lib/classes/eik-config.js [hunk 1]
```diff
@@ -43,11 +43,14 @@ module.exports = class EikConfig {
         this[_tokens] = new Map(tokens);
 
         assert(
+            // @ts-ignore
```
Comment (wkillerud) [inline lib/classes/eik-config.js:46]: Super if you can do a search and replace `ts-ignore` with `ts-expect-error` so we get notified when they can be removed 🙏

File: lib/classes/local-file-location.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 /**
  * @type {(value: unknown, message?: string) => asserts value}
  */
-const assert = require('assert');
-const { join, extname, isAbsolute } = require('path');
-const mime = require('mime-types');
+import assert from 'node:assert';
+import { join, extname, isAbsolute } from 'node:path';
+// @ts-ignore
+import mime from 'mime-types';
```
Comment (wkillerud) [inline lib/classes/local-file-location.js:7]: This should hopefully let us remove this `ts-ignore`
```
npm install --save-dev @types/mime-types
```

**Review by wkillerud — APPROVED** [review]
A couple of type nits, but looks good to me!

**Comment by digitalsadhu** [comment]
There was an existing refactor of the whole code base on next which was stale but looks very much like the direction we want to head in so I've moved that to a branch called "future-refactor" and set next back to match current main. This PR is now against next and I'll merge it into that and continue work updating dependencies from there

**Pull request merged**